core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksStore.java

```diff
@@ -159,19 +171,22 @@ private void createDb() throws RocksDBException {
    */
   public synchronized void writeToCheckpoint(OutputStream output)
       throws IOException, InterruptedException {
-    LOG.info("Backing up rocks database to {}", mDbBackupPath);
+    LOG.info("Creating rocksdb checkpoint at {}", mDbCheckpointPath);
     long startNano = System.nanoTime();
+
     CheckpointOutputStream out = new CheckpointOutputStream(output, CheckpointType.ROCKS);
-    try (BackupEngine backupEngine = BackupEngine.open(Env.getDefault(),
-        new BackupableDBOptions(mDbBackupPath).setMaxBackgroundOperations(4))) {
-      backupEngine.createNewBackup(mDb, true);
-      backupEngine.purgeOldBackups(1);
+    try {
+      // createCheckpoint requires that the directory not already exist.
```

Comment:
what sets it to `ROCKS_CHECKPOINT`? What happens to backup engine?

Reply:
backup engine is no longer being used in any way